components/workflow-mgt/org.wso2.carbon.identity.workflow.mgt/src/main/java/org/wso2/carbon/identity/workflow/mgt/WorkflowManagementServiceImpl.java

```diff
@@ -455,10 +618,20 @@ public boolean areTwoEntitiesRelated(Entity entity1, Entity entity2) throws
      * @throws InternalWorkflowException
      */
     @Override
-    public boolean isEventAssociated(String eventType) throws InternalWorkflowException {
+    public boolean isEventAssociated(String eventType) throws WorkflowException {
 
-        List<WorkflowAssociation> associations = workflowRequestAssociationDAO.getWorkflowAssociationsForRequest(eventType, CarbonContext
+        List<WorkflowListener> workflowListenerList =
+                WorkflowServiceDataHolder.getInstance().getWorkflowListenerList();
+        for (WorkflowListener workflowListener : workflowListenerList) {
+            workflowListener.doPreIsEventAssociated(eventType);
+        }
+        List<WorkflowAssociation> associations = workflowRequestAssociationDAO.getWorkflowAssociationsForRequest
+                (eventType, CarbonContext
```

Comment:
Thanks for pointing this, fixed with https://github.com/wso2/carbon-identity/pull/1675